examples/linalg/src/lib.rs

```diff
@@ -3,9 +3,9 @@ use numpy::{IntoPyArray, PyArray2, PyReadonlyArray2};
 use pyo3::{exceptions::PyRuntimeError, pymodule, types::PyModule, PyResult, Python};
 
 #[pymodule]
-fn rust_linalg(_py: Python, m: &PyModule) -> PyResult<()> {
+fn rust_linalg<'py>(_py: Python<'py>, m: &'py PyModule) -> PyResult<()> {
```

Comment:
Can these (and others) be anonymous? i.e. `fn rust_linalg(_py: Python<'_>, m: &'_PyModule) -> PyResult<()> {`?

Reply:
They can, but I decidedly tried to keep the GIL lifetime named `'py` instead of keeping it anonymous where that would be possible. I think if we do add these for documentation, we should go all the way.